xgcm/sgrid.py

```diff
@@ -0,0 +1,230 @@
+from collections import OrderedDict
+
+
+def assert_valid_sgrid(ds):
+    """Verify that the dataset meets SGRID conventions
+    We know this if the dataset has 'SGRID' listed in its conventions
+
+    Parameters
+    ----------
+    ds : xarray.dataset
+
+    Returns
+    ----------
+    True/False : boolean
+    """
+
+    if "SGRID" in ds.Conventions:
+        return True
+    elif "sgrid" in ds.Conventions:
+        return True
+    elif "Sgrid" in ds.Conventions:
+        return True
+
+    return False
```

Comment:
Minor style nitpick:

```suggestion
    elif "Sgrid" in ds.Conventions:
        return True
    else:
        return False
```

Reply:
Tidied this up, and as part of the issue above (checking for conventions) now involving nested ifs I think the final return is now required, rather than being inside a final else.